### PR TITLE
=str - fixing TODO about Source.wrap not having to re-wrap Sources

### PR DESCRIPTION
**Pull request by viktorklang**
Boy Scouting

EDIT: Also added optional wrapping of javadsl.{Source, Sink, BidiFlow, Flow} and scaladsl.{Source, Sink, BidiFlow, Flow}

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala [hunk 1]
```diff
@@ -202,8 +202,10 @@ object Source extends SourceApply {
    * A graph with the shape of a source logically is a source, this method makes
    * it so also in type.
    */
-  // TODO optimize if no wrapping needed
-  def wrap[T, M](g: Graph[SourceShape[T], M]): Source[T, M] = new Source(g.module)
+  def wrap[T, M](g: Graph[SourceShape[T], M]): Source[T, M] = g match {
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: Javadsl needs this, too. Also, Flow and Sink and BidiFlow has wrap too.

Comment (viktorklang) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala]: I can most definitely fix those too, I just saw this one and acted :)

**Comment by viktorklang** [comment]
@drewhk done!

**Comment by drewhk** [comment]
LGTM. (The impact will be minor though, since the heavyweight object (the underlying Module) was already just shared, and Source et. al. were just lightweight wrappers around it.)

**Comment by viktorklang** [comment]
@drewhk Of course, but removes a tiny cut to prevent the death by a thousand :)

**Comment by viktorklang** [comment]
Can I merge this?

**Comment by drewhk** [comment]
yes

**Pull request merged**